packages/block-editor/src/components/block-parent-selector/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockType } from '@wordpress/blocks';
+import { Button } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockIcon from '../block-icon';
+
+/**
+ * Block parent selector component, displaying the hierarchy of the
+ * current block selection as a single icon to "go up" a level.
+ *
+ * @return {WPElement} Parent block selector.
+ */
+export default function BlockParentSelector() {
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+	const { parentBlockType, firstParentClientId } = useSelect( ( select ) => {
+		const {
+			getBlockName,
+			getBlockParents,
+			getSelectedBlockClientId,
+		} = select( 'core/block-editor' );
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const _parents = getBlockParents( selectedBlockClientId );
+		const _firstParentClientId = _parents[ _parents.length - 1 ];
```

Comment:
No need to prefix `parents` with an underscore since it's not conflicting with anything in the upper scope:
```suggestion
		const parents = getBlockParents( selectedBlockClientId );
		const _firstParentClientId = parents[ parents.length - 1 ];
```